Need to add one step in quick start procedure of About Logging 6.0 documentation

prithvipatil97 opened the pull request:
- Need to add one step in the Quick Start point of About Logging 6.0 documentation.
- Here is the documentation link: https://docs.openshift.com/container-platform/4.17/observability/logging/logging-6.0/log6x-about.html#quick-start
- Here Step 2 is mentioned as "Create a LokiStack custom resource (CR) in the openshift-logging namespace:"
- But before creating `LokiStack` custom resource (CR), it is necessary to create an `object storage secret`.
- Without creating a secret we can not create `Lokistack` customer resource.
- We need to mention that `object storage secret` name in the `LokiStack custom resource (CR)` under `spec.storage.secret.name` section.
- So it is required to add this step in our documentation.
- Hence adding Step 2 in the documentation:

-----------------
2.  Create a secret with the details to access to an existing object storage bucket. AWS is used in the following example: 

~~~
$ oc create secret generic logging-loki-s3 \
  --from-literal=bucketnames="<bucket_name>" \
  --from-literal=endpoint="<aws_bucket_endpoint>" \
  --from-literal=access_key_id="<aws_access_key_id>" \
  --from-literal=access_key_secret="<aws_access_key_secret>" \
  --from-literal=region="<aws_region_of_your_bucket>" \
  -n openshift-logging
~~~
------------------

<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->

RHOCP-4.18, RHOCP-4.17, RHOCP-4.16, RHOCP-4.15, RHOCP-4.14

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

https://issues.redhat.com/browse/OBSDOCS-1346

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

https://86648--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/logging/logging-6.0/log6x-about.html

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

prithvipatil97 commented:
/label peer-review-needed

xenolinux commented on observability/logging/logging-6.0/log6x-about.adoc:
```diff
@@ -43,6 +43,19 @@ Logging includes extensive validation rules and default values to ensure a smoot
 
 . Install the `OpenShift Logging` and `Loki` Operators from OperatorHub.
 
+. Create a secret with the details to access to an existing object storage bucket. AWS is used in the following example: 
++
```
```suggestion
. Create a secret to access an existing object storage bucket:  
+
.Example command for AWS
```

prithvipatil97 replied:
Hello @xenolinux ,
Thank you very much for reviewing this PR and providing some suggestion.

Sure.
Let me try to commit the suggested changes and squash the changes.

Regards,
Prithviraj Patil

xenolinux reviewed (commented):
A tiny nit; otherwise LGTM

Gentle reminder for QE review.

prithvipatil97 commented:
Hello @xenolinux ,
Thanks for your suggestions.

As suggested by you I performed commit changes.
But I am facing an issue while Squash commits. 

~~~
# git clone https://github.com/prithvipatil97/openshift-docs
fatal: destination path 'openshift-docs' already exists and is not an empty directory

# git clone --depth 1 https://github.com/prithvipatil97/openshift-docs
Cloning into 'openshift-docs'...
remote: Enumerating objects: 11078, done.
remote: Counting objects: 100% (11078/11078), done.
remote: Compressing objects: 100% (10043/10043), done.
remote: Total 11078 (delta 1506), reused 3231 (delta 998), pack-reused 0 (from 0)
Receiving objects: 100% (11078/11078), 75.90 MiB | 439.00 KiB/s, done.
Resolving deltas: 100% (1506/1506), done.

# git checkout patch-17
error: pathspec 'patch-17' did not match any file(s) known to git
~~~

I tried some actions to resolve this issue, but it is not working.

Let me create a New PR with suggested new changes, and will link that new PR with this PR.
I hope it will be fine.

Kindly confirm.

Regards,
Prithviraj Patil

xenolinux commented:
> Hello @xenolinux , Thanks for your suggestions.
> 
> As suggested by you I performed commit changes. But I am facing an issue while Squash commits.
> 
> ```
> # git clone https://github.com/prithvipatil97/openshift-docs
> fatal: destination path 'openshift-docs' already exists and is not an empty directory
> 
> # git clone --depth 1 https://github.com/prithvipatil97/openshift-docs
> Cloning into 'openshift-docs'...
> remote: Enumerating objects: 11078, done.
> remote: Counting objects: 100% (11078/11078), done.
> remote: Compressing objects: 100% (10043/10043), done.
> remote: Total 11078 (delta 1506), reused 3231 (delta 998), pack-reused 0 (from 0)
> Receiving objects: 100% (11078/11078), 75.90 MiB | 439.00 KiB/s, done.
> Resolving deltas: 100% (1506/1506), done.
> 
> # git checkout patch-17
> error: pathspec 'patch-17' did not match any file(s) known to git
> ```
> 
> I tried some actions to resolve this issue, but it is not working.
> 
> Let me create a New PR with suggested new changes, and will link that new PR with this PR. I hope it will be fine.
> 
> Kindly confirm.
> 
> Regards, Prithviraj Patil

@prithvipatil97 I don't think you are following the correct steps. see https://spaces.redhat.com/display/OSDOCS/Keeping+your+local+repository+in+sync+with+upstream+changes

prithvipatil97 commented:
Hello @xenolinux ,
Thanks for your response and for sharing the correct step.

I followed all the steps. 
But currently, it is failing at the last step, with the git push command. Getting following error:
~~~
remote: Permission to prithvipatil97/openshift-docs.git denied to prithvipatil97.
fatal: unable to access 'https://github.com/prithvipatil97/openshift-docs/': The requested URL returned error: 403
~~~

Here are steps followed:
~~~
pripatil@pripatil-thinkpadt14sgen2i:~$ git clone https://github.com/prithvipatil97/openshift-docs
Cloning into 'openshift-docs'...
remote: Enumerating objects: 573405, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 573405 (delta 2), reused 0 (delta 0), pack-reused 573401 (from 1)
Receiving objects: 100% (573405/573405), 1.05 GiB | 1.82 MiB/s, done.
Resolving deltas: 100% (389441/389441), done.
~~~
~~~
pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git checkout patch-17
branch 'patch-17' set up to track 'origin/patch-17'.
Switched to a new branch 'patch-17'
~~~
~~~
pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git rebase -i HEAD~2
[detached HEAD fdf3e73b68] Need to add one step in quick start procedure of About Logging 6.0 documentation
 Author: Prithviraj Patil <116709298+prithvipatil97@users.noreply.github.com>
 Date: Thu Jan 2 22:12:32 2025 +0530
 Committer: Prithviraj Patil <pripatil@pripatil-thinkpadt14sgen2i.punetw6.csb>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly. Run the
following command and follow the instructions in your editor to edit
your configuration file:

    git config --global --edit

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 1 file changed, 14 insertions(+)
Successfully rebased and updated refs/heads/patch-17.
~~~
~~~
pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git log -5 --pretty=format:'%h - %an, %ar : %s'
fdf3e73b68 - Prithviraj Patil, 8 days ago : Need to add one step in quick start procedure of About Logging 6.0 documentation
9fa27f3d93 - Servesha Dudhgaonkar, 10 days ago : Merge pull request #85711 from sbeskin-redhat/CNV_46268_Replace_running_with_runStrategy
7830941439 - Joe Aldinger, 11 days ago : Merge pull request #85972 from JoeAldinger/OCPBUGS-45841
284d366f83 - Servesha Dudhgaonkar, 11 days ago : Merge pull request #86619 from xenolinux/cp-x86
2d92228834 - xenolinux, 2 weeks ago : OSDOCS#13000: Clarify platforms for CP and nodepools in IBM-Z/P
~~~
~~~
pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git push origin HEAD --force
Username for 'https://github.com': prithvipatil97
Password for 'https://prithvipatil97@github.com': 
remote: Permission to prithvipatil97/openshift-docs.git denied to prithvipatil97.
fatal: unable to access 'https://github.com/prithvipatil97/openshift-docs/': The requested URL returned error: 403
~~~

I am using the correct username and password.
Instead of a password, I tried with a Personal Access Token as well. But it is still not working.

Here is the repository URL:
~~~
pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git remote -v
origin	https://github.com/prithvipatil97/openshift-docs (fetch)
origin	https://github.com/prithvipatil97/openshift-docs (push)
~~~

Could you please help me here.
I think I have a permission issue here.


Regards,
Prithviraj Patil

xenolinux commented:
> Hello @xenolinux , Thanks for your response and for sharing the correct step.
> 
> I followed all the steps. But currently, it is failing at the last step, with the git push command. Getting following error:
> 
> ```
> remote: Permission to prithvipatil97/openshift-docs.git denied to prithvipatil97.
> fatal: unable to access 'https://github.com/prithvipatil97/openshift-docs/': The requested URL returned error: 403
> ```
> 
> Here are steps followed:
> 
> ```
> pripatil@pripatil-thinkpadt14sgen2i:~$ git clone https://github.com/prithvipatil97/openshift-docs
> Cloning into 'openshift-docs'...
> remote: Enumerating objects: 573405, done.
> remote: Counting objects: 100% (4/4), done.
> remote: Compressing objects: 100% (4/4), done.
> remote: Total 573405 (delta 2), reused 0 (delta 0), pack-reused 573401 (from 1)
> Receiving objects: 100% (573405/573405), 1.05 GiB | 1.82 MiB/s, done.
> Resolving deltas: 100% (389441/389441), done.
> ```
> 
> ```
> pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git checkout patch-17
> branch 'patch-17' set up to track 'origin/patch-17'.
> Switched to a new branch 'patch-17'
> ```
> 
> ```
> pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git rebase -i HEAD~2
> [detached HEAD fdf3e73b68] Need to add one step in quick start procedure of About Logging 6.0 documentation
>  Author: Prithviraj Patil <116709298+prithvipatil97@users.noreply.github.com>
>  Date: Thu Jan 2 22:12:32 2025 +0530
>  Committer: Prithviraj Patil <pripatil@pripatil-thinkpadt14sgen2i.punetw6.csb>
> Your name and email address were configured automatically based
> on your username and hostname. Please check that they are accurate.
> You can suppress this message by setting them explicitly. Run the
> following command and follow the instructions in your editor to edit
> your configuration file:
> 
>     git config --global --edit
> 
> After doing this, you may fix the identity used for this commit with:
> 
>     git commit --amend --reset-author
> 
>  1 file changed, 14 insertions(+)
> Successfully rebased and updated refs/heads/patch-17.
> ```
> 
> ```
> pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git log -5 --pretty=format:'%h - %an, %ar : %s'
> fdf3e73b68 - Prithviraj Patil, 8 days ago : Need to add one step in quick start procedure of About Logging 6.0 documentation
> 9fa27f3d93 - Servesha Dudhgaonkar, 10 days ago : Merge pull request #85711 from sbeskin-redhat/CNV_46268_Replace_running_with_runStrategy
> 7830941439 - Joe Aldinger, 11 days ago : Merge pull request #85972 from JoeAldinger/OCPBUGS-45841
> 284d366f83 - Servesha Dudhgaonkar, 11 days ago : Merge pull request #86619 from xenolinux/cp-x86
> 2d92228834 - xenolinux, 2 weeks ago : OSDOCS#13000: Clarify platforms for CP and nodepools in IBM-Z/P
> ```
> 
> ```
> pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git push origin HEAD --force
> Username for 'https://github.com': prithvipatil97
> Password for 'https://prithvipatil97@github.com': 
> remote: Permission to prithvipatil97/openshift-docs.git denied to prithvipatil97.
> fatal: unable to access 'https://github.com/prithvipatil97/openshift-docs/': The requested URL returned error: 403
> ```
> 
> I am using the correct username and password. Instead of a password, I tried with a Personal Access Token as well. But it is still not working.
> 
> Here is the repository URL:
> 
> ```
> pripatil@pripatil-thinkpadt14sgen2i:~/openshift-docs$ git remote -v
> origin	https://github.com/prithvipatil97/openshift-docs (fetch)
> origin	https://github.com/prithvipatil97/openshift-docs (push)
> ```
> 
> Could you please help me here. I think I have a permission issue here.
> 
> Regards, Prithviraj Patil

Hey @prithvipatil97 To resolve these errors, we need to jump in the meeting and it could take time. I am also on PTO and will be back Wed. I think you can consider creating another PR by closing this PR. Add the merge review label again since the peer review is done. You can mention this PR link in the PR description. Thanks!

prithvipatil97 commented:
Hello @xenolinux ,
Thank you very much for your response.

Sure.
Let me create a new PR with new suggested changes and will link it to this PR for reference.

And we will close this PR.

Regards,
Prithviraj Patil

prithvipatil97 commented:
Created a new PR:
Linked to https://github.com/openshift/openshift-docs/pull/86970

prithvipatil97 commented:
New PR has been linked to this PR.
Hence closing this PR.
